src/Donations/LegacyListeners/DispatchGiveInsertPayment.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Give\Donations\LegacyListeners;
+
+use Give\Donations\Models\Donation;
+use Give\Donors\Models\Donor;
+use Give\Helpers\Hooks;
+use Give\PaymentGateways\DataTransferObjects\GiveInsertPaymentData;
+
+class DispatchGiveInsertPayment
+{
+    /**
+     * @unreleased
+     *
+     * @param  Donation  $donation
+     * @return void
+     */
+    public function __invoke(Donation $donation)
+    {
+        /** @var Donor $donor */
+        $donor = $donation->donor()->get();
+
+        $giveInsertPaymentData = GiveInsertPaymentData::fromArray([
+            'price' => $donation->getMinorAmount(),
+            'formTitle' => $donation->formTitle,
+            'formId' => $donation->formId,
+            'priceId' => give_get_price_id($donation->formId, $donation->getMinorAmount()->getAmount()),
+            'date' => $donation->createdAt,
+            'donorEmail' => $donor->email,
+            'purchaseKey' => $donation->purchaseKey,
+            'currency' => $donation->currency,
+            'paymentGateway' => $donation->gateway,
+            'userInfo' => [
+                'id' => $donor->id,
+                'firstName' => $donor->firstName,
+                'lastName' => $donor->lastName,
+                'title' => $donor->prefix,
+                'email' => $donor->email
+            ],
+        ]);
+
+         /**
+         * @deprecated
+         */
+        Hooks::dispatch('give_insert_payment', $donation->id, $giveInsertPaymentData->toArray());
```

Comment:
I'm okay with that, the main reason I made this was to log all the do_actions after they are called for sanity sakes.  I don't think that's necessary for production but I could see it being helpful during development in the future. So maybe we can wrap that logger to check for `WP_DEBUG_LOG`. What do you think?

Reply:
So long as we're using `Log::info()` under the hood it's already only logging those with debug mode enabled. Honestly, it may be worth adding a `Log::debug()` method that checks for something like `GIVE_DEBUG`. What do you think?

Reply:
It's true. I just think there's some logging that's truly informative to everyone (such as a webhook being received), and then there's stuff like logging _every_ action that can easily flood the log.

Reply:
Yeah I agree

Reply:
Do you want me to want to this in this PR, or a seperate one?

Reply:
It's so tiny, just go ahead and do it here.